app/controllers/stock_status_controller.rb

```diff
@@ -0,0 +1,73 @@
+class StockStatusController < ApplicationController
+  
+  def index
+  	if( params[:hub].present? && params[:program].present? && params[:commodity].present? )
+  		@result = ActiveRecord::Base.connection
+		.exec_query('SELECT hubs.name AS hub, commodities.name AS commodity, donors.name AS donor, programs.name AS program, 
+			EPI.expected, RPI.received, DiPI.dispatched, DePI.delivered, (DiPI.dispatched - DePI.delivered)  AS git
+		FROM
+		(Select hub_id, program_id, commodity_id, donor_id, ABS(SUM(quantity)) AS expected from posting_items 
+			WHERE (journal_id = 2 OR journal_id = 4 OR journal_id = 7) AND account_id = 1
```

Comment:
If we could make the journals/accounts CONSTANTS - instead of writing the actual journal id or account id #